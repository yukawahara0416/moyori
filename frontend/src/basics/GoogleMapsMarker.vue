<template>
  <div>
    <gmap-marker
      v-for="(m, id) in markers"
      :key="id"
      :icon="m.icon"
      :position="m.position"
      :title="m.name"
      :zIndex="m.zIndex"
      @click="
        changeIcon(m, id)
        panTo(m.position)
      "
    />
  </div>
</template>

<script>
import { mapGetters } from 'vuex'

export default {
  computed: {
    ...mapGetters(['markers']),

    cache: {
      get() {
        return this.$store.getters.cache
      },
      // markerの選択状態をCardが監視
      set(value) {
        this.$store.dispatch('updateCurrentMarker', value)
      }
    }
  },

  methods: {
    changeIcon(marker, id) {
      this.$store.dispatch('changeIcon', { marker: marker, id: id })
    },

    panTo(pos) {
      this.$emit('pan-to', pos)
    }
  }
}
</script>
