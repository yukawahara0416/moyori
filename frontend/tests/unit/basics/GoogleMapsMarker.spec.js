import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import GoogleMapsMarker from '@/basics/GoogleMapsMarker.vue'
import markerStore from '@/store/modules/marker.js'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper
let store
let state
let actions

beforeEach(() => {
  state = {
    spots: [
      { marker: { name: 'hoge', icon: 'foo' } },
      { marker: { name: 'fuga', icon: 'bar' } }
    ],
    cache: { id: 11, icon: 'puge' }
  }

  actions = {
    setCurrentMarker: jest.fn()
  }

  store = new Vuex.Store({
    modules: {
      markerStore: {
        state,
        getters: markerStore.getters,
        actions
      }
    }
  })

  wrapper = shallowMount(GoogleMapsMarker, {
    localVue,
    store,
    stubs: ['gmap-marker']
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('emit', () => {
  it('pan-to', () => {
    const pos = { lat: 123, lng: 45 }
    wrapper.vm.panTo(pos)
    expect(wrapper.emitted('pan-to')).toBeTruthy()
    expect(wrapper.emitted('pan-to')[0][0]).toEqual(pos)
  })
})

describe('getters', () => {
  it('markers', () => {
    expect(wrapper.vm.markers).toEqual(state.markers)
  })

  it('cache', () => {
    expect(wrapper.vm.cache).toEqual(state.cache)
  })
})

describe('actions', () => {
  it('setCurrentMarker', () => {
    wrapper.vm.setCurrentMarker()
    expect(actions.setCurrentMarker).toHaveBeenCalled()
  })
})

describe('template', () => {
  it('v-for', () => {
    expect(wrapper.findAll('gmap-marker-stub').length).toBe(2)
  })
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
