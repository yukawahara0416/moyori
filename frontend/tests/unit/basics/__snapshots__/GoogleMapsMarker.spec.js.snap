// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div>
  <gmap-marker-stub
    animation="4"
    icon="foo"
    name="hoge"
  />
  <gmap-marker-stub
    animation="4"
    icon="bar"
    name="fuga"
  />
</div>
`;
